src/tracer/MOM_generic_tracer.F90

```diff
@@ -572,6 +611,17 @@ subroutine MOM_generic_tracer_column_physics(h_old, h_new, ea, eb, fluxes, Hml,
     enddo ; enddo
     sosga = global_area_mean(surface_field, G, unscale=US%S_to_ppt)
 
+    mld_pha(:,:) = 0.0          
+    if (.not.CS%mld_pha_calc) then
+      mld_pha(:,:) = CS%mld_pha_val
+    else
+      if (CS%mld_pha_use_delta_rho) then
+        call diagnoseMLDbyDensityDifference(-1, h_old, tv, CS%mld_pha_drho, G, GV, US, CS%diag, CS%mld_pha_href, id_ref_z=-1, id_ref_rho=-1, MLD_out=mld_pha)
+      elseif (CS%mld_pha_use_delta_eng) then
+        call diagnoseMLDbyEnergy((/-1, -1, -1/), h_old, tv, G, GV, US, (/CS%mld_pha_deng, CS%mld_pha_deng, CS%mld_pha_deng/), CS%diag, MLD_out=mld_pha)
+      endif
```

Comment:
@theresa-morrison, do you think it would be better to add a warning message here in case users accidentally set both `CS%mld_pha_use_delta_rho` and `CS%mld_pha_use_delta_eng` while still having `CS%mld_pha_calc` turned on? I believe our `all_same` check should catch this at bgc side, but providing a more informative message would clarify that not only is `CS%mld_pha_calc` turned on, but either `CS%mld_pha_use_delta_rho` or `CS%mld_pha_use_delta_eng` needs to be enabled as well in order to get spatial varying `mld_pha`.

Reply:
yeah, that is a good idea. 
If mld_pha_calc is false, the settings of mld_pha_use_delta_rho and mld_pha_use_delta_eng are actually not read at all. I will modify the logic around where the parameters are read to give a warning to keep things simple here.